Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/StoreOperation/ADLSGen2FSOperation.java

```diff
@@ -76,12 +85,33 @@ public ADLSGen2FSOperation(@NotNull HttpObservable http) {
 
         this.flushReqParamsBuilder = new ADLSGen2ParamsBuilder()
                 .setAction("flush");
+
+        this.setAclReqParams = new ADLSGen2ParamsBuilder()
+                .setAction("setAccessControl")
+                .build();
+
+        this.getAclReqParams = new ADLSGen2ParamsBuilder()
+                .setAction("getAccessControl")
+                .build();
     }
 
     public Observable<Boolean> createDir(String dirpath) {
         HttpPut req = new HttpPut(dirpath);
-        return http.executeReqAndCheckStatus(req, 201, this.createDirReqParams)
+        Observable<Boolean> res = http.executeReqAndCheckStatus(req, 201, this.createDirReqParams)
                 .map(ignore -> true);
+        return  res;
+    }
+
+    public Observable<Boolean> setAcl(String dirPath, HeaderGroup additionHeaderGroup) {
```

Comment:
Let the 2nd parameter to be posix formats, such as 664, 755. The type may be int or enum. Don’t forget to validate the value.